OpenShift deployment fails over HTTP POST request to service monitor URL

### Describe the bug

I'm unable to deploy application to OpenShift. I've tried to revert https://github.com/quarkusio/quarkus/pull/37709 which made no difference.

### Expected behavior

Deployment works fine, as does with Quarkus 3.6.3.

### Actual behavior

Deployment fails with following error:

```
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (build) on project many-extensions: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	[error]: Build step io.quarkus.kubernetes.deployment.KubernetesDeployer#deploy threw an exception: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://reducted:6443/apis/monitoring.coreos.com/v1/servicemonitors. Message: Not Found.
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.KubernetesClientException.copyAsCause(KubernetesClientException.java:238)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:507)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:524)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleCreate(OperationSupport.java:340)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.handleCreate(BaseOperation.java:753)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.handleCreate(BaseOperation.java:97)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.CreateOnlyResourceOperation.create(CreateOnlyResourceOperation.java:42)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.utils.internal.CreateOrReplaceHelper.createOrReplace(CreateOrReplaceHelper.java:53)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.createOrReplace(BaseOperation.java:313)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.createOrReplace(BaseOperation.java:97)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.extension.ResourceAdapter.createOrReplace(ResourceAdapter.java:124)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesDeployer.deployResource(KubernetesDeployer.java:242)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesDeployer.lambda$deploy$4(KubernetesDeployer.java:202)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesDeployer.deploy(KubernetesDeployer.java:201)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesDeployer.deploy(KubernetesDeployer.java:142)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://reducted:6443/apis/monitoring.coreos.com/v1/servicemonitors. Message: Not Found.
[2023-12-19T10:56:01.449Z] 10:56:01,288 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:660)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:640)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.assertResponseCode(OperationSupport.java:589)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.lambda$handleResponse$0(OperationSupport.java:549)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$10(StandardHttpClient.java:140)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.http.ByteArrayBodyHandler.onBodyDone(ByteArrayBodyHandler.java:52)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.fabric8.kubernetes.client.vertx.VertxHttpRequest.lambda$null$1(VertxHttpRequest.java:122)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:277)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:259)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.HttpEventHandler.handleEnd(HttpEventHandler.java:76)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.HttpClientResponseImpl.handleEnd(HttpClientResponseImpl.java:250)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.lambda$new$0(Http1xClientConnection.java:446)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:255)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:134)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.handleEnd(Http1xClientConnection.java:710)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:76)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.impl.ContextBase.execute(ContextBase.java:290)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.Http1xClientConnection.handleResponseEnd(Http1xClientConnection.java:946)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.Http1xClientConnection.handleHttpMessage(Http1xClientConnection.java:816)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.http.impl.Http1xClientConnection.handleMessage(Http1xClientConnection.java:780)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1471)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1334)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1383)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[2023-12-19T10:56:01.449Z] 10:56:01,289 INFO  mvn: [ERROR] -> [Help 1]
```

### How to Reproduce?

Steps to reproduce the behavior:

1. `git clone git@github.com:quarkus-qe/quarkus-test-suite.git`
3. `cd quarkus-test-suite/super-size/many-extensions`
4. `git revert 03fd09d60ebd693ce122d21d9b5d584883caed33`
5. `mvn clean verify -Dopenshift -Dit.test=OpenShiftUsingExtensionManyExtensionsIT`


Also following tests are failing:

```
io.quarkus.ts.many.extensions.OpenShiftUsingExtensionDockerBuildStrategyManyExtensionsIT
io.quarkus.ts.many.extensions.ServerlessExtensionDockerBuildStrategyOpenShiftManyExtensionsIT
io.quarkus.ts.many.extensions.ServerlessExtensionOpenShiftManyExtensionsIT
```

### Output of `uname -a` or `ver`

Fedora 38

### Output of `java -version`

openjdk version "21.0.1" 2023-10-17 LTS

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3

### Additional information

OpenShift 4.14 and 4.11.

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  annotations:
    app.openshift.io/vcs-uri: <<unknown>>
    app.quarkus.io/commit-id: 1f08d0d4c800bd032d284a8e20bd69737ae8c154
    app.quarkus.io/vcs-uri: git@github.com:quarkus-qe/quarkus-test-suite.git
    app.quarkus.io/build-timestamp: 2023-12-19 - 23:21:02 +0000
  labels:
    app.kubernetes.io/name: app
    app.kubernetes.io/version: 1.0.0-SNAPSHOT
    scenarioId: OpenShiftUsingExtensionManyExtensionsIT-1703028032464
    tsLogWatch: app
    app.kubernetes.io/managed-by: quarkus
    app.openshift.io/runtime: quarkus
  name: app
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: app
      app.kubernetes.io/version: 1.0.0-SNAPSHOT
  endpoints:
    - scheme: http
      targetPort: 8080
      path: /api/q/metrics
      interval: 10s
      honorLabels: true
```

I run the test suite locally and got: 

```
11:14:01,709 INFO  mvn: Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://api.qshift.snowdrop.dev:6443/apis/monitoring.coreos.com/v1/servicemonitors. Message: Not Found.
```

I think that the URL used to post the resource is wrong (it requires a namespace). 
I will look into it.

FWIW, including the error message in the description is immensely helpful.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -164,7 +164,7 @@
         <kotlin.coroutine.version>1.7.3</kotlin.coroutine.version>
         <azure.toolkit-lib.version>0.27.0</azure.toolkit-lib.version>
         <kotlin-serialization.version>1.6.2</kotlin-serialization.version>
-        <dekorate.version>4.1.1</dekorate.version> <!-- Please check with Java Operator SDK team before updating -->
+        <dekorate.version>4.1.2</dekorate.version> <!-- Please check with Java Operator SDK team before updating -->
         <maven-invoker.version>3.2.0</maven-invoker.version>
         <awaitility.version>4.2.0</awaitility.version>
         <jboss-logmanager.version>3.0.4.Final</jboss-logmanager.version>

